zcash_client_sqlite/src/wallet.rs

```diff
@@ -1170,14 +1169,14 @@ pub fn put_sent_note<'a, P: consensus::Parameters>(
     tx_ref: i64,
     output_index: usize,
     account: AccountId,
-    to: &PaymentAddress,
+    to: &Recipient,
```

Comment:
- The type here is invalid, as this allows a transparent address to be passed in.
- This is a change to a public deprecated API, making the deprecation meaningless. We should not change the public deprecated API, and instead either map to the new API internally (as was done below for `put_sent_utxo`) or remove the deprecated API.

Reply:
I guess that this was a copy-paste error in https://github.com/zcash/librustzcash/commit/cb6a99384059b0bacff56a0e1e4de498b0a610bd

Reply:
Oh, wait. No. It's that the `sent_notes` table is actually also being used to track UTXOs, and the intermediate APIs are just obscuring this fact: https://github.com/zcash/librustzcash/blob/main/zcash_client_sqlite/src/wallet.rs#L1230-L1288

So the correct thing, as of right now, is to get rid of basically all of this and instead maybe rename the `sent_notes` table to something that more accurately reflects its role, and I guess re-create the views so that they're showing the right information.

Reply:
Fixed in https://github.com/zcash/librustzcash/pull/659/commits/179e22b76d8a34eff7cfc995683dcf34692816ac